src/newRequestSequence.js

```diff
@@ -5,66 +5,47 @@
 import {
-  REQUEST,
-  SUCCESS,
   FAILURE,
   FINALLY,
+  REQUEST,
+  SUCCESS,
 } from './ActionTypes';
-import type { SequenceAction, SequenceActionCreator } from './getSequenceAction';
-import type { SwitchCaseMatcher } from './getSwitchCaseMatcher';
-import type { SequenceReducer } from './getSequenceReducer';
-
-type RequestSequence = {
-  REQUEST :string;
-  SUCCESS :string;
-  FAILURE :string;
-  FINALLY :string;
-  (...args :any[]) :SequenceAction;
-  request :SequenceActionCreator;
-  success :SequenceActionCreator;
-  failure :SequenceActionCreator;
-  finally :SequenceActionCreator;
-  baseType :string;
-  case :SwitchCaseMatcher;
-  reducer :SequenceReducer;
-};
+import { isNonEmptyString, randomStringId } from './utils/utils';
+import type { SequenceAction } from './getSequenceAction';
+import type { RequestSequence } from './types';
 
 /*
  * TODO: need logger
  */
 
-function isValidId(value :any) :boolean {
-
-  return Object.prototype.toString.call(value) === STRING_TAG && value.length > 0;
-}
-
 export default function newRequestSequence(baseType :string) :RequestSequence {
 
-  const requestSequenceActionType :string = `${baseType}/${REQUEST}`.toUpperCase();
-  const successSequenceActionType :string = `${baseType}/${SUCCESS}`.toUpperCase();
-  const failureSequenceActionType :string = `${baseType}/${FAILURE}`.toUpperCase();
-  const finallySequenceActionType :string = `${baseType}/${FINALLY}`.toUpperCase();
-
-  const sequences :{[key :string] :Object} = {};
+  const sequences = {};
+  const rsActionTypes = {
+    [REQUEST]: `${baseType}/${REQUEST}`.toUpperCase(),
+    [SUCCESS]: `${baseType}/${SUCCESS}`.toUpperCase(),
+    [FAILURE]: `${baseType}/${FAILURE}`.toUpperCase(),
+    [FINALLY]: `${baseType}/${FINALLY}`.toUpperCase(),
+  };
 
-  const triggerActionCreator :RequestSequence = (triggerValue :any) :SequenceAction => {
+  const requestSequence = <V>(triggerValue ?:V) :SequenceAction<V> => {
     const id :string = randomStringId();
     sequences[id] = {
       requestCalled: false,
       successCalled: false,
       failureCalled: false,
-      request: (requestValue :any) => getSequenceAction(id, requestSequenceActionType, requestValue),
-      success: (successValue :any) => getSequenceAction(id, successSequenceActionType, successValue),
-      failure: (failureValue :any) => getSequenceAction(id, failureSequenceActionType, failureValue),
-      finally: (finallyValue :any) => getSequenceAction(id, finallySequenceActionType, finallyValue)
+      request: <RequestValue>(value ?:RequestValue) => getSequenceAction(id, rsActionTypes[REQUEST], value),
+      success: <SuccessValue>(value ?:SuccessValue) => getSequenceAction(id, rsActionTypes[SUCCESS], value),
+      failure: <FailureValue>(value ?:FailureValue) => getSequenceAction(id, rsActionTypes[FAILURE], value),
+      finally: <FinallyValue>(value ?:FinallyValue) => getSequenceAction(id, rsActionTypes[FINALLY], value),
     };
+    // $FlowFixMe
```

Comment:
🤔 What's the issue here?

Reply:
![Screen Shot 2019-12-11 at 12 36 28 PM](https://user-images.githubusercontent.com/440299/70658461-dfec0f00-1c12-11ea-96a2-0b7395375201.png)

Reply:
It's basically complaining that `<V>` defined in the `RequestSequence` type is different from the `<V>` defined locally in this function, on line 32...

```
const requestSequence = <V>(triggerValue ?:V) :SequenceAction<V> => {
  ...
};
```

I think the fix is probably to parameterize `RequestSequence`, i.e. ``RequestSequence<V>`, but that was not the goal of this PR, and I'm not totally sure if that will fix it.